### ISSUE TITLE
Problems with Opaque struct serialization

### ISSUE THREAD

**Issue by SkipMike** [assert-value/assert_value_elixir#4]
0.8.1 is looking to be a great release and a huge step forward for this library. However, there are still some small problems with the serialization. 

For instance, if you do
```
thing = Flow.from_enumerable([:a]) |> Flow.map(fn x -> IO.inspect(x) end)
assert_value thing
```
It's going to print out 
```
%Flow{
  operations: [{:mapper, :map, [#Function<0.78071283/1 in ....
}
```
However, the `#` that is before Function is going to comment out the rest of that line so it's never going to actually match.

I'm not sure what can be done about it, but I'm hoping that some sort of auto-canonicalization of opaque types can be done.

**Comment by SkipMike** [comment]
I'm sorry, I should've read the whole readme again. I just saw that you already knew about this issue.

**Comment by smetana** [comment]
Thanks. I have reopened this issue. This is a valid problem. We will look into it.

**Comment by smetana** [comment]
Released 0.8.2 with better check and error message about not-serializable values
